controller/controller.go

```diff
@@ -229,11 +230,45 @@ func (c *HAProxyController) handlePath(namespace *Namespace, ingress *Ingress, r
 // handleEndpointIP processes the IngressPath related endpoints and makes corresponding backend servers configuration in HAProxy
 func (c *HAProxyController) handleEndpointIP(namespace *Namespace, ingress *Ingress, rule *IngressRule, path *IngressPath, service *Service, backendName string, newBackend bool, endpoints *Endpoints, ip *EndpointIP) (needReload bool) {
 	needReload = false
+	weight := c.osArgs.DefaultWeight
+	podStatus := EMPTY
+	if ip.TargetRef.Kind == "Pod" {
+		pod := c.cfg.Pods[string(ip.TargetRef.UID)]
+		if pod == nil {
+			if DEBUG_API {
+				log.Printf("Pod %s with UID %s in namespace %s not found in cache",
+					ip.TargetRef.Name,
+					string(ip.TargetRef.UID),
+					namespace.Name)
+			}
+		} else {
+			podStatus = pod.Status
+			annotation, err := GetValueFromAnnotations("weight", pod.Annotations)
+			if err != nil {
+				if !strings.Contains(err.Error(), "does not exist") {
+					utils.LogErr(err)
+				}
+			} else {
+				weight, err = strconv.ParseInt(annotation.Value, 10, 64)
+				if err != nil {
+					utils.LogErr(err)
+				}
+			}
+			switch podStatus {
+			case MODIFIED:
+				fallthrough
+			case DELETED:
+				fallthrough
+			case ADDED:
+				needReload = true
+			}
```

Comment:
what do you think about replacing this switch with
```
switch podStatus {
	case MODIFIED, DELETED, ADDED:
		needReload = true
}
```
fallthrough is ok, but I think its more readable without it

Reply:
Done